patched for building on Solaris.  Potentially applicable to be sent upstream

--- gstreamer-1.4.5/gst/gsturi.c	2014-07-21 01:20:56.000000000 -0700
+++ gstreamer-1.4.5/gst/gsturi.c	2015-06-02 12:13:19.084628371 -0700
@@ -639,11 +639,11 @@
  * Returns: the #GstURIType of the URI handler.
  * Returns #GST_URI_UNKNOWN if the @handler isn't implemented correctly.
  */
-guint
+GstURIType
 gst_uri_handler_get_uri_type (GstURIHandler * handler)
 {
   GstURIHandlerInterface *iface;
-  guint ret;
+  GstURIType ret;
 
   g_return_val_if_fail (GST_IS_URI_HANDLER (handler), GST_URI_UNKNOWN);
 
